src/main/java/org/apache/flink/scheduler/benchmark/SchedulerBenchmarkUtils.java

```diff
@@ -0,0 +1,245 @@
+import org.apache.flink.runtime.executiongraph.JobInformation;
+import org.apache.flink.runtime.executiongraph.NoOpExecutionDeploymentListener;
+import org.apache.flink.runtime.executiongraph.TaskExecutionStateTransition;
+import org.apache.flink.runtime.executiongraph.failover.RestartAllStrategy;
+import org.apache.flink.runtime.executiongraph.failover.flip1.partitionrelease.RegionPartitionReleaseStrategy;
+import org.apache.flink.runtime.executiongraph.restart.NoRestartStrategy;
+import org.apache.flink.runtime.io.network.partition.NoOpJobMasterPartitionTracker;
+import org.apache.flink.runtime.io.network.partition.ResultPartitionType;
+import org.apache.flink.runtime.jobgraph.DistributionPattern;
+import org.apache.flink.runtime.jobgraph.JobGraph;
+import org.apache.flink.runtime.jobgraph.JobVertex;
+import org.apache.flink.runtime.jobgraph.JobVertexID;
+import org.apache.flink.runtime.jobgraph.ScheduleMode;
+import org.apache.flink.runtime.jobmaster.LogicalSlot;
+import org.apache.flink.runtime.jobmaster.TestingLogicalSlotBuilder;
+import org.apache.flink.runtime.jobmaster.slotpool.SlotProvider;
+import org.apache.flink.runtime.scheduler.DefaultScheduler;
+import org.apache.flink.runtime.shuffle.NettyShuffleMaster;
+import org.apache.flink.runtime.taskmanager.TaskExecutionState;
+import org.apache.flink.runtime.testingUtils.TestingUtils;
+import org.apache.flink.runtime.testtasks.NoOpInvokable;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Predicate;
+
+/**
+ * Utilities for runtime benchmarks.
+ */
+public class SchedulerBenchmarkUtils {
+
+	public static List<JobVertex> createDefaultJobVertices(
+			int parallelism,
+			DistributionPattern distributionPattern,
+			ResultPartitionType resultPartitionType) {
+
+		List<JobVertex> jobVertices = new ArrayList<>();
+
+		final JobVertex source = new JobVertex("source");
+		source.setInvokableClass(NoOpInvokable.class);
+		source.setParallelism(parallelism);
+		jobVertices.add(source);
+
+		final JobVertex sink = new JobVertex("sink");
+		sink.setInvokableClass(NoOpInvokable.class);
+		sink.setParallelism(parallelism);
+		jobVertices.add(sink);
+
+		sink.connectNewDataSetAsInput(source, distributionPattern, resultPartitionType);
+
+		return jobVertices;
+	}
+
+	public static JobGraph createJobGraph(
+			List<JobVertex> jobVertices,
+			ScheduleMode scheduleMode,
+			ExecutionMode executionMode) throws IOException {
+
+		final JobGraph jobGraph = new JobGraph(jobVertices.toArray(new JobVertex[0]));
+
+		jobGraph.setScheduleMode(scheduleMode);
+		ExecutionConfig executionConfig = new ExecutionConfig();
+		executionConfig.setExecutionMode(executionMode);
+		jobGraph.setExecutionConfig(executionConfig);
+
+		return jobGraph;
+	}
+
+	public static ExecutionGraph createExecutionGraph(
```

Comment:
Maybe we can just use those utils in `SchedulerTestingUtils` in fink repo?

Reply:
> Maybe we can just use those utils in `SchedulerTestingUtils` in fink repo?

Okay, I'll check if there are any utils can be replaced by existing utils in Flink repo. Once finished, I'll update the progress here.

Reply:
@zhuzhurk I've update the dependency of Flink to version 1.13-SNAPSHOT. The utils are fixed, too. Would you mind reviewing it again if you got free time? Thank you 🙏